### PR TITLE
Update dependencies

### PR DESCRIPTION
**Pull request by vgeddes**
Need to update ethereum-types for compatibility with latest Substrate.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -7,8 +7,7 @@ license = "Apache-2.0"
 edition = "2018"
 
 [dependencies]
-ethereum-types = { version = "0.10.0", default-features = false }
-primitive-types = { version = "0.8", default-features = false, features = ["rlp"] }
```
Comment (Rizziepit) [inline Cargo.toml:11]: Does the new ethereum-types include primitive-types with feature rlp?

**Review by Rizziepit — APPROVED** [review]
I assume you've run the parachain tests. Please make sure you include the WASM test like so: 
```
cd parachain/primitives/ethereum
cargo install -f wasm-bindgen-cli  # this avoids a binary version issue
cargo test --target wasm32-unknown-unknown
```

**Comment by vgeddes** [comment]
> I assume you've run the parachain tests. Please make sure you include the WASM test like so:

All good! the parachain unit tests all pass (after upgrading them to the latest substrate):

The WASM test passes too:
```
cargo +nightly test --target wasm32-unknown-unknown
...
Set timeout to 20 seconds...
running 1 test

test artemis_ethereum::ethashproof::tests::hashimoto_breakdown_11550000_wasm ... ok

test result: ok. 1 passed; 0 failed; 0 ignored
```

File: Cargo.toml (hunk 1 above)
Comment (vgeddes) [reply Cargo.toml:11]: It seems ethereum-types by default just includes primitive-types without rlp:
https://github.com/paritytech/parity-common/blob/master/ethereum-types/Cargo.toml#L14

Also, the direct dependency on primitive-types wasn't necessary as it we weren't importing `primitive_types` anywhere.

Anyway the code compiles and the tests pass so we should be in the clear:
```
   Compiling hex-literal v0.2.1
   Compiling ethash v0.5.0 (/home/vincent/code/ethash)
    Finished test [unoptimized + debuginfo] target(s) in 0.87s
     Running target/debug/deps/ethash-67870bdcf6197ef6

running 1 test
test tests::hashimoto_should_work ... test tests::hashimoto_should_work has been running for over 60 seconds
test tests::hashimoto_should_work ... ok

test result: ok. 1 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out; finished in 121.66s

   Doc-tests ethash

running 0 tests

test result: ok. 0 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.00s
```

**Pull request merged**